vizro-core/src/vizro/actions/_action_manager/_action_manager.py

```diff
@@ -0,0 +1,40 @@
+"""The action manager handles creation of all required actions callbacks."""
+
+from vizro.actions._action_manager._build_action_loop_callbacks import _build_action_loop_callbacks
```

Comment:
Shall we move the action manager to vizro.managers? Or did you keep it here because the first resulted in lots of circular import issues?

Reply:
Hm, I think we should rename it to `ActionsLoopEngine` or something, because it is not a manager. It doesn't really manage anything as it is a one way communication.

Reply:
I prefer ActionManager as I think it also fits better with the front-end world and the syntax they use, even if it's just calling static functions. I think at some point we do actually want to extend the responsibility of the ActionManager e.g. for registering custom action functions and then population the relevant mapping files. 

Just adding some examples here and their definition of the action manager, which I still find quite fitting:

- https://doc.babylonjs.com/typedoc/classes/BABYLON.ActionManager
"Action Manager manages all events to be triggered on a given mesh or the global scene."

- https://doc.qt.io/qtcreator-extending/core-actionmanager.html#details
"The action manager is the central bookkeeper of actions and their shortcuts and layout. It is a singleton containing mostly static functions."

Reply:
Hm, I am still unsure. Yes it is used in other code bases, but so are other class names if one googles them.

The reason why I am unsure is because we seem to be picking a name that stands for a pattern in our code base. I find it confusing, because you attach expectations of certain behaviour (global variable, stores something, violates state) with that name.

If we are to extend this, we can always change the name, so why not pick a name that is more suitable now, and adjust that when we add things later. After all it is private (unlike our other managers, which are public). But then again, because it is private, the name also doesn't matter hugely, so it is not the most important question I think

Reply:
It doesn't matter very because it's private, but have to agree with @maxschulz-COL. If we all agree, I would rename it to the `ActionsLoopEngine` or something similar.

Reply:
I really don't like the Engine as a description because it has nothing to do with an engine for me😄 

 If you guys want to rename to something different than ActionManager - then maybe (I asked chat gpt):
- ActionBuilder
- ActionFactory
- ActionSequenceFactory
- ActionLoopFactory
- ActionComponentCreator

Reply:
My preferences are:
 - ActionLoop (calling like: ActionLoop.build or ActionLoop.create_app_callbacks),
 - ActionLoopFactory (calling like: ActionLoopFactory.build, ActionLoopFactory.create_app_callbacks)

Reply:
Either of those two would be fine for me. If we think we align enough with this: https://refactoring.guru/design-patterns/factory-method , then lets go for the second, if not, lets do the first

Reply:
Then let's go with ActionLoop

Reply:
Sounds like a plan 👍.